intellij/src/saros/intellij/SarosComponent.java

```diff
@@ -36,35 +43,39 @@ public SarosComponent() {
           SarosComponent.class.getClassLoader().getResourceAsStream("saros.properties");
 
       if (sarosProperties == null) {
-        LogLog.warn(
-            "could not initialize Saros properties because "
-                + "the 'saros.properties' file could not be found on the "
-                + "current JAVA class path");
+        StatusLogger.getLogger()
+            .warn(
+                "could not initialize Saros properties because "
+                    + "the 'saros.properties' file could not be found on the "
+                    + "current JAVA class path");
       } else {
         System.getProperties().load(sarosProperties);
         sarosProperties.close();
       }
     } catch (Exception e) {
-      LogLog.error("could not load saros property file 'saros.properties'", e);
+      StatusLogger.getLogger().error("could not load saros property file 'saros.properties'", e);
     }
 
     IntellijApplicationLifecycle.getInstance().start();
   }
 
-  private void loadLoggers() {
-    final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
-
+  private void setupLoggers() {
     try {
-      // change the context class loader so Log4J will find
-      // the SarosLogFileAppender
-      Thread.currentThread().setContextClassLoader(SarosComponent.class.getClassLoader());
+      final String logDir = PathManager.getLogPath() + File.separator + "SarosLogs";
+      final boolean isDebugMode = Boolean.getBoolean("saros.debug");
+      final Level logLevel = isDebugMode ? Level.ALL : Level.INFO;
+
+      // set path to configuration file to be used during reconfiguration
+      System.setProperty("log4j.configurationFile", LOG4J2_CONFIG_FILENAME);
+      // make arguments accessible in the log configuration file
+      MainMapLookup.setMainArguments("logDir", logDir, "logLevel", logLevel.name());
 
-      PropertyConfigurator.configure(
-          SarosComponent.class.getClassLoader().getResource("saros.log4j.properties"));
+      // trigger reconfiguration with new properties and config file
+      LoggerContext context = (LoggerContext) LogManager.getContext(false);
```

Comment:
Isn't this https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/LogManager.html#getContext-java.lang.ClassLoader-boolean-java.net.URI- the call you want to use ?

Reply:
What do you think is the advantage of handing a class loader? It could be necessary if we had custom appenders/plugins etc., but we don't use it.

Reply:
You simple avoid using the System.property call. Have not found a better method yet that only takes the configuration file URL.

Reply:
Ok, I can try it. I already tried [`setConfigLocation`](https://logging.apache.org/log4j/2.x/log4j-core/apidocs/org/apache/logging/log4j/core/LoggerContext.html#setConfigLocation-java.net.URI-), but it seemed like the config was not available in all loggers (only the first initialization log messages were logged). Using the property worked for all loggers. Probably, I have to call `updateLoggers`.

Reply:
Seems weird as `setConfigLocation` states that it does a reconfigure and at this point no additional loggers should be created yet.

Reply:
> but it seemed like the config was not available in all loggers (only the first initialization log messages were logged

As described in the last commit message, the issue was that log4j2 created a logger context for each class loader (each bundle had its own context and only the eclipse messages were logged not the core or stf messages).